lib/elixir/lib/enum.ex

```diff
@@ -2259,14 +2266,12 @@ defmodule Enum do
     do: []
   def take_random(first..first, count) when is_integer(count) and count >= 1,
     do: [first]
-  def take_random(first..last, 1) when first > last,
-    do: take_random(last..first, 1)
-  def take_random(first..last, 1),
-    do: [random_index(last - first) + first]
+  def take_random(_.._ = range, 1),
+    do: [random(range)]
```

Comment:
We need to either change behaviour for all ranges or stop special casing `1` inside `take_random`. I think it should be simpler to change behaviour for all ranges. Note this problem does not exist for Enum.random (because it is always `1` and we don't provide any guarantee between `random` and `take_random`).